modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/TableColumnHeaderTest.java

```diff
@@ -0,0 +1,146 @@
+
+public class TableColumnHeaderTest {
+
+    private MyTableColumnHeader tableColumnHeader;
+
+    @Before
+    public void beforeTest() {
+        tableColumnHeader = null;
+    }
+
+    @Test
+    public void test_resizeColumnToFitContent() {
+        ObservableList<Person> model = FXCollections.observableArrayList(
+                new Person("Humphrey McPhee", 76),
+                new Person("Justice Caldwell", 30),
+                new Person("Orrin Davies", 30),
+                new Person("Emma Wilson", 8)
+        );
+        TableColumn<Person, String> column = new TableColumn<>("Col ");
+        column.setCellValueFactory(new PropertyValueFactory<Person, String>("firstName"));
+        TableView<Person> tableView = new TableView<>(model) {
+            @Override
+            protected Skin<?> createDefaultSkin() {
+                return new TableViewSkin(this) {
+                    @Override
+                    protected TableHeaderRow createTableHeaderRow() {
+                        return new TableHeaderRow(this) {
+                            @Override
+                            protected NestedTableColumnHeader createRootHeader() {
+                                return new NestedTableColumnHeader(null) {
+                                    @Override
+                                    protected TableColumnHeader createTableColumnHeader(TableColumnBase col) {
+                                        tableColumnHeader = new MyTableColumnHeader(column);
+                                        return col == null || col.getColumns().isEmpty() || col == getTableColumn() ?
+                                                tableColumnHeader :
+                                                new NestedTableColumnHeader(col);
+
+                                    }
+                                };
+                            }
+                        };
+                    }
+                };
+            }
+        };
+
+        tableView.getColumns().add(column);
+
+        Toolkit tk = Toolkit.getToolkit();
+
+        Scene scene = new Scene(tableView);
+        Stage stage = new Stage();
+        stage.setScene(scene);
+        stage.setWidth(500);
+        stage.setHeight(400);
+        stage.centerOnScreen();
+        stage.show();
+
+        tk.firePulse();
+
+        double width = column.getWidth();
+        tableColumnHeader.resizeCol();
+        assertEquals("Width must be the same",
+                width, column.getWidth(), 0.001);
+
+        EventType<TableColumn.CellEditEvent<Person, String>> eventType = TableColumn.editCommitEvent();
+        column.getOnEditCommit().handle(new TableColumn.CellEditEvent<Person, String>(
+                tableView, new TablePosition<Person, String>(tableView, 0, column), (EventType) eventType, "This is a big text inside that column"));
+        tableColumnHeader.resizeCol();
+        assertTrue("Column width must be greater",
+                width < column.getWidth());
+
```

Comment:
feels a bit unspecific (though I can't think of how to do it better - the internal measuring is ... doohoo ;)

what I usually do in such a case is to take this as a first step, then change content back to original and again trigger a resize: now the current width should be the same as the initial width.

Reply:
But what if the resizeCol never worked and the value never changed at all?

Reply:
that's what the first step guards against :)

Reply:
Allright, allright, I'll add the initial width Assert then